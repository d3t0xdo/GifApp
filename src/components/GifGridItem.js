import React from 'react';
import 'animate.css';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {

	return (
		<div className="card">
			<img src={ url } alt={ title } className="card_image"></img>	
			<div className="card_overlay">
				<div className="card_header">
					<div className="card_header_text">
						<h3 className="card__title">{ title }</h3>
					</div>
				</div>
			</div>
		</div>
	)
}

GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired
}