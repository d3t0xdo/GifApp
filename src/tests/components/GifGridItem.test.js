import React from 'react'
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Test of the component GifGridItem', () => {
	
	test('Test of the return in the component', () => {
		
		const testUrl = "https://google.com";
		const testTitle = "Title of the component";

		const wrapper = shallow( <GifGridItem />);
		expect( wrapper ).toMatchSnapshot();

	});
	
});
