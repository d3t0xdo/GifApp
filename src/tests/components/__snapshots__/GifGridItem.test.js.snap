// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of the component GifGridItem Test of the return in the component 1`] = `
<div
  className="card"
>
  <img
    className="card_image"
  />
  <div
    className="card_overlay"
  >
    <div
      className="card_header"
    >
      <div
        className="card_header_text"
      >
        <h3
          className="card__title"
        />
      </div>
    </div>
  </div>
</div>
`;
